### PR TITLE
Restructure event directories

### LINKED ISSUE #123: Rework seasonal branding structure
The current setup we have for seasonal branding is out of date. Even the name doesn't work anymore. In order to work well with https://github.com/python-discord/bot/issues/1431, we need to reorganize these assets and standardize on a single approach.

## Wait, what's this for?
Read https://github.com/python-discord/bot/issues/1431 before you read this. Basically we want every PyDis event (or maybe we should call it something else? Holiday? Doodle?) to contain all the information needed in order for the bot to automatically manage branding assets.

So, it's Valentines day, and the bot changes the icon, changes the discord banner, and posts a message into #changelog with a short explanation about what the icon represents. That's the end product.

## Okay, so what changes do we need to make?
Well, currently we have the `seasonal` folder, and we've got the `events` folder. These two should be merged into a single folder, and we should get rid of anything that doesn't fit our new structure.

For each event, we'll need a structure roughly like this:
```
├── events/
│ ├── valentines/
│ │ ├── server_icons/
│ │ │ ├── love_heart.gif
│ │ │ ├── beating_heart.gif
│ │ │ ├── valentines.png
│ │ ├── misc/
│ │ │ ├── anything_can_be_in_here.txt
│ │ │ ├── the_bot_will_ignore_it.png
│ │ │ ├── could_be_useful_for_major_events.zip
│ │ ├── banner.png
│ │ ├── meta.md
```

This is pretty similar to how the `seasonal` folders were organized back in https://github.com/python-discord/branding/pull/46, with the exception of the `meta.md` file.

## So how would the `meta.md` look?
Here's an example of how it might look, although you're free to iterate on this as you see fit.
```md
---
start_date: 2021-12-01
end_date: 2021-12-25
repeats: annually  # For one-time events, we could put 'never'.
---
**Advent of Code**
The server icon and banner have been changed to the **Advent of Code** branding. You can learn more about Advent of Code [on their website](https://adventofcode.com).
```

## What about all the stuff we're currently missing?
Since we do not have banners for all the stuff inside `events`, and we don't have `meta.md` for any of the events, this additional material should be created in this PR. If that's not feasible, we should just get rid of the event, and we can reimplement it with a fully compatible PR later on.

## How do we test and validate submissions?
It would be fantastic if we were able to validate the contents of a PR into this folder using CI, but I think this should probably be a separate issue that we can tackle later. I imagine a piece of CI that would ensure dates were valid, ensure the dates didn't conflict with another event, and ensure that all the required files were there.

*edited by lemonsaurus*

### LINKED ISSUE #127: Create Easter themed banner
The [Easter season](https://github.com/python-discord/branding/tree/master/seasonal/easter) currently lacks a banner asset.

For a simple solution, the contribution can simply adapt the [Evergreen banner](https://github.com/python-discord/branding/blob/master/seasonal/evergreen/banner.png) and apply the colours used by the Easter logo.

However, there's plenty of room for creativity. For example, the banner could incorporate usual Easter motives. :rabbit: :egg: :chocolate_bar: 

Feel free to discuss ideas here or in `#media-branding` on Discord.

### PR DESCRIPTION
**Pull request by kwzrd**
Closes #123.
Closes #127.

---

This PR implements the changes proposed in #123.

The `events` directory (previously `seasonal`) is restructured and configured with `meta.md` files to be used by [Python's new branding manager](https://github.com/python-discord/bot/pull/1463). A validation script is added, which will run in CI and validate event configuration to ensure that a broken setup doesn't reach the `main` branch.

# What has changed?

The changes are not too dramatic - the new `events` directory is very similar in structure to what we had before. The new addition are `meta.md` files, which use [frontloaded YAML](https://assemble.io/docs/YAML-front-matter.html) to provide each event with a description and a duration. These files are fetched and read by the Python bot.

The exact structure that events must comply with in order to pass validation is documented in the new [`events/README.md`](https://github.com/kwzrd/pydis-branding/blob/kwzrd/events-rework/events/README.md). I will therefore not go into too much detail, as I assume that reviewers will read the README anyway.

Branding for past one-time events (i.e. jams) is not included in the `events` directory. I've placed those assets into a parallel `jams` directory. It is assumed that the branding manager will be used to manage jam assets in the future, but it does not make sense to include past events as they will never be applied again.

# Validation & CI

I have added a validation script that lives in the `events` directory. It's a few lines of code that will run through the directory and verify that each event is correctly set up, and that there are no collisions between events. It depends on the YAML parser declared in the requirements file. The script exits with a non-0 code if it detects errors.

image

image

image

I have also added a GH Actions workflow that runs this script on pull requests and new commits on `main`. However, it seems that since I am working from a fork, we will not be able to run the workflow until the PR is merged into the upstream. Reviewers can run the script locally (see the events README for instructions). I have tested and verified that it fails the workflow if errors are found.

PyDis projects use a custom workflow for sending status embeds to the `#mod-log` channel, reporting whether CI passed. Because I have struggled a little bit with understanding how this workflow works, and didn't want to blindly copy-paste it without being able to test, I'd prefer for it to be added later (directly in upstream, and probably by someone more experienced). So for now the CI runs and validates events, but does not produce a cute status embed.

# Assets

This PR adds an Easter banner, which was previously missing. It is a simple colour edit of the evergreen SVG:

image

# Event descriptions

With the help of @dawnofmidnight we have added a short description for each event. The descriptions are pulled by Python and presented as embeds:

image

Each description links Wikipedia or similar for more information. I assume there may be some discussion regarding the descriptions. I wasn't entirely sure on how to approach them.

# Calendar

Quick overview of current configuration, formatted as `event_name[start_day-end_day]`. Both bounds inclusive.

```
February: Valentines[1-15], BHM[16-29]
March: StPatrick[10-18]
April: Easter[1-30]
June: Pride[1-30]
October: Hacktoberfest[1-31]
November: Diwali[1-7]
December: Christmas[1-25], NewYear[26-31]
```

Python-side, a calendar overview can be produced on-demand from the fetched events:

image

I'd recommend that reviewers also read the [Python-side PR description](https://github.com/python-discord/bot/pull/1463).

### CODE REVIEW

**Comment by kwzrd** [comment]
This PR will now also add the missing Easter banner (#127). It's a colour edit of the evergreen one to match the Easter logo:

image

PR includes both the SVG base & a rendered PNG.

**Comment by kwzrd** [comment]
:pray:

Ready for review!

**Review by lemonsaurus — DISMISSED** [review]
This looks excellent to me. Very well done.

File: events/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Events
+
+This directory contains branding for events celebrated by Python Discord. Events are automatically discovered by the [Python](https://github.com/python-discord/bot/) bot and have their assets applied to the Discord guild when active. As such, event directories must follow a specific structure.
+
+## Event directory structure
+
+In order for an event directory to be valid, it has to contain the following assets.
```
Comment (Shivansh-007) [inline events/README.md:7]:
```suggestion
For an event directory to be valid, it has to contain the following assets.
```
`in order for` seems a bit wordy here, it is grammatically correct, but keeping it concise is better.

**Review by Shivansh-007 — APPROVED** [review]
Looks good! Just 2 grammar changes (can be ignored) and it would be perfect!

**Comment by kwzrd** [comment]
@lemonsaurus Thank you for the review!

@Shivansh-007 Thanks, applied in f8c4282.

**Review by Shivansh-007 — APPROVED** [review]
purrrfect

**Review by lemonsaurus — APPROVED** [review]
Let's do this.
